### PR TITLE
Add --version options to all scripts and update slurm submission scripts

### PR DESCRIPTION
**Pull request by clairecporter**
This PR adds a --version option to all scripts and prints the version in the log.  It also adds in the slurm submission scripts and an option to submit to the slurm scheduler.  The `--pbs` and `--slurm` options were replaced with a `--scheduler` option with `pbs` and `slurm` as choices. 

When this is ready to merge, I think I should make a new version, since the CLI options have changed and are no longer back-compatible (unless we make some dummy options that do the same thing as the --scheduler option).

### CODE REVIEW

**Comment by clairecporter** [comment]
@ehusby Can you please take a look at this PR.  No need for comprehensive testing.  But I'd appreciate your eyes on the changes I made before I roll them in.

File: package_setsm_tiles.py [hunk 1]
```diff
@@ -384,7 +385,7 @@ def build_archive(raster, scratch, args):
                             if lyr is not None:
 
                                 for field_def in utils.TILE_DEM_ATTRIBUTE_DEFINITIONS_BASIC + utils.DEM_ATTRIBUTE_DEFINITION_RELVER:
-                                    if field_def.ftype == ogr.OFTDateTime and ogr_driver_str in ['ESRI Shapefile']:
+                                    if field_def.ftype == ogr.OFTDateTime:
```
[the remark this answers is not in the document]
Comment (clairecporter) [reply package_setsm_tiles.py:388]: Shapefile does not support DateTime, so I think it's safe.

[the remark this answers is not in the document]
Comment (ehusby) [reply package_setsm_tiles.py:388]: At the end of our discussion, the idea is to write out any DateTime fields as string fields in shapefiles.

Maybe we should prep the `package_setsm.py` script with this same conversion methodology. Currently all of the strip index fields are still strings so the code wouldn't have any effect, but it'd be good for increasing awareness.

[the remark this answers is not in the document]
Comment (clairecporter) [reply package_setsm_tiles.py:388]: As we discussed, I'll change this to a String field and also manually set the width.

[the remark this answers is not in the document]
Comment (clairecporter) [reply package_setsm_tiles.py:388]: Good point. I'll do both. And make the width wide enough for a timestamp.

**Comment by clairecporter** [comment]
@ehusby take a look at the OFTDateTime sections when the format is shp and see if it looks good to you.  The existing tests pass, though I did not add one for date length.

**Comment by ehusby** [comment]
Looks good!

**Comment by clairecporter** [comment]
@ehusby I made a set of changes to add back-compatible `--pbs` and `--slurm` options.  Plus I refactored a bit to make the repeated code less unwieldy and more maintainable.

File: build_stac_tree.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def main():
         parser.error("Source directory or file does not exist: %s" %args.src)
 
     if args.validate:
-        pass
+        import pystac
```
Comment (clairecporter) [inline build_stac_tree.py]: Ignore this change as it's actually rolling back a mistake I made earlier

**Review by ehusby — APPROVED** [review]
Looks good!

**Pull request merged**